src/dc_smtp.c

```diff
@@ -208,6 +208,15 @@ int dc_smtp_connect(dc_smtp_t* smtp, const dc_loginparam_t* lp)
 				}
 				r = mailesmtp_auth_sasl(smtp->etpan, "PLAIN", hostname, NULL, NULL, NULL, lp->send_user, lp->send_pw, NULL);
 			}
+
+			/*
+			* There is a quirk with *some* Mailservers configuration in which PLAIN auth will not work if not 
+			* properly announced by the server, but it will happily proceed if we assume it is allowed.
+			*/
+			if (r == MAILSMTP_NO_ERROR) {
```

Comment:
shouldn't this read `if (r != MAILSMTP_NO_ERROR)`?

Reply:
you are right, this shouldn't be fixing the bug I was seeing